packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -71,19 +117,93 @@ class TimeInput extends React.Component {
     this.handleFocus = this.handleFocus.bind(this);
     this.handleHourBlur = this.handleHourBlur.bind(this);
     this.handleMinuteBlur = this.handleMinuteBlur.bind(this);
+    this.handleMeridiemChange = this.handleMeridiemChange.bind(this);
+    this.handleMeridiemInputKeyDown = this.handleMeridiemInputKeyDown.bind(this);
+    this.meridiemFocus = this.meridiemFocus.bind(this);
+    this.meridiemBlur = this.meridiemBlur.bind(this);
+    this.handleMeridiemButtonChange = this.handleMeridiemButtonChange.bind(this);
+
+    let hour = TimeUtil.splitHour(value);
+    let meridiem = null;
+
+    if (props.variant === FORMAT_12_HOUR) {
+      this.anteMeridiem = context.intl.formatMessage({ id: 'Terra.timeInput.am' });
+      this.postMeridiem = context.intl.formatMessage({ id: 'Terra.timeInput.pm' });
+
+      if (hour) {
+        const parsedHour = this.parseTimeTwelveHourTime(hour, true);
+        hour = parsedHour.hourString;
+        meridiem = parsedHour.meridiem;
+      } else {
+        meridiem = this.anteMeridiem;
+      }
+    }
+
+    this.state = {
+      hour,
+      minute: TimeUtil.splitMinute(value),
+      isFocused: false,
+      meridiem,
```

Comment:
Should we check if its format before saving the meridian state? Wound't need to track this a 24 clock

Reply:
The hard part with this is that the component is made so that it can switch between a 12 hour and 24 hour time, meaning we would be switching between setting and deleting this key. After reading around on react from others in the community, it sounds like it's best practice to instead set the key to undefined if you anticipate needing to set and unset it. https://stackoverflow.com/questions/32884780/how-can-i-remove-an-attribute-from-a-reactjs-components-state-object.

I'll change the default value in the constructor to undefined to better reflect this.